ha/util/actuator_req_manager.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import time
+import json
+import ast
+
+from cortx.utils.conf_store import Conf
+from ha.core.system_health.const import NODE_MAP_ATTRIBUTES
+from ha.execute import SimpleCommand
+from ha.const import ACTUATOR_SCHEMA, ACTUATOR_ATTRIBUTES, ACTUATOR_RESP_RETRY_COUNT, ACTUATOR_RESP_WAIT_TIME
+from ha.const import _DELIM
+from ha import const
+from cortx.utils.log import Log
+from ha.const import PVTFQDN_TO_NODEID_KEY
+
+from ha.core.system_health.system_health import SystemHealth, SystemHealthManager
+from ha.core.system_health.const import EVENT_SEVERITIES
+from ha.core.error import ClusterManagerError
+from ha.util.message_bus import MessageBus
+from ha.core.config.config_manager import ConfigManager
+
+
+class ActuatorManager:
+
+    def __init__(self):
+
+        self._conf_store = ConfigManager.get_confstore()
+        self._execute = SimpleCommand()
+        self._machine_id = self._get_machine_id()
+        self._uuid = None
+        self._is_resp_received = False
+        self._encl_shutdown_successful = False
+
+    def _generate_uuid(self):
+        """
+        Generate as uuid a random number.
+        This will be populated in uuid field in actuator request
+        same uuid is expected in actuatore response
+        """
+        self._uuid = str(int(time.time()))
+
+    def _get_machine_id(self):
+        command = "cat /etc/machine-id"
```

Comment:
we can move this to ha utils and can reuse at multiple places

Reply:
Done